### PR TITLE
Encrypted storage

### PR DESCRIPTION
**Pull request by emilbayes**
Note this will break `client.list_keys` and `client.send_transaction` if the wallet is locked at the moment

### CODE REVIEW

File: web-extension/common/test/encrypted-storage.spec.js [hunk 1]
```diff
@@ -2,44 +2,85 @@ import EncryptedStorage from '../lib/encrypted-storage.js'
...
+    await expect(encryptedStorage.create('passphrase')).rejects.toThrow('Storage already exists')
+  })
+
+  test('Create storage twice with overwrite', async () => {
+    const encryptedStorage = new EncryptedStorage(new Map())
+
+    await encryptedStorage.create('passphrase')
 
-    await expect(encryptedStorage.get('foo')).rejects.toThrow('Storage is not open')
+    await encryptedStorage.create('passphrase', true)
```
Comment (dexturr) [inline web-extension/common/test/encrypted-storage.spec.js:84]: no assertions on a test?

**Comment by dalebennett1992** [comment]
Looks like we have e2e failures that are consistent on chrome and firefox, which would suggest something is genuinely broken ;)

**Review by dexturr — APPROVED** [review]

**Review by dalebennett1992 — CHANGES_REQUESTED** [review]
need to investigate these failures before we send down. Will take a look

File: web-extension/common/test/admin-ns.spec.js [hunk 2]
```diff
@@ -2,12 +2,15 @@ import initAdminServer from '../backend/admin-ns.js'
 import { WalletCollection } from '../backend/wallets.js'
```
Comment (dalebennett1992) [inline web-extension/common/test/admin-ns.spec.js:2]: mentioned in slack but thought I would chuck it in here too. Would like to see a few more tests just to exercise the positive and negative paths.. in particular these things

- sad path for admin.update passphrase to check it gives the encryption error
- happy path for admin.unlock
- sad path for admin.unlock to check it gives 'encryption not initialised'
- admin.lock

Comment (emilbayes) [reply web-extension/common/test/admin-ns.spec.js:2]: These should be covered in `b42f872`

File: package.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
   },
   "scripts": {
     "dev:chrome": "web-ext run --source-dir=build/chrome --target=chromium --start-url=examples/sample-dapp/index.html",
-    "dev:firefox": "web-ext run --source-dir=build/firefox --devtools --target=firefox-desktop --start-url=examples/sample-dapp/index.html",
+    "dev:firefox": "web-ext run --source-dir=build/firefox --firefox=firefoxdeveloperedition --start-url=examples/sample-dapp/index.html",
```
Comment (emilbayes) [inline package.json:33]: @dalebennett1992 I inadvertently committed this, maybe that breaks firefox tests

**Comment by dexturr** [comment]
superseded by https://github.com/vegaprotocol/vegawallet-browser/pull/174
